README.md

```diff
@@ -179,6 +177,23 @@ class User {
 
 Queries that transform the encrypted column wont work because transformers and subscribers operate outside of the DBMS.
 
+### What alogorithm should I use?
+
+Unless you need to maintain compatibility with an older system (or you know exactly what you're doing),
+you should use "aes-256-gcm" for the mode.
```

Comment:
Unfortunately, I think GCM or CTR modes require the set of IVs to be unique. There can be no overlap, basically.

The code currently generates IVs randomly if not provided. It doesn't guarantee uniqueness.

For GCM mode, perhaps we should allow `options.iv` to be a function where library users can provide a function that guarantees uniqueness.

Until then, CBC might be safer.

Reply:
CBC requires unique IVs, too. All modes of operation for AES to (with the possible exception of SIV which isn't available in Node anyway).

If the IVs are 16-bytes long and generated with a Cryptographically Secure Pseudo Random Number Generator (CSPRNG) then that is perfectly fine. The `randomBytes` function (that is used internally now) meets this requirement.

Providing a function to generate the IV would work but puts a lot of responsibility on the developer and could make mistakes more likely.